std/os/path.zig

```diff
@@ -35,38 +35,61 @@ pub fn isSep(byte: u8) bool {
 
 /// Naively combines a series of paths with the native path seperator.
 /// Allocates memory for the result, which must be freed by the caller.
-pub fn join(allocator: *Allocator, paths: ...) ![]u8 {
-    if (is_windows) {
-        return joinWindows(allocator, paths);
-    } else {
-        return joinPosix(allocator, paths);
+
+pub fn join(allocator: *Allocator, paths: []const []const u8) ![]u8 {
+    assert(paths.len >= 1);
+    var total_paths_len: usize = paths.len; // 1 sep per path
+    {
+        var path_i: usize = 0;
+        while (path_i < paths.len) : (path_i += 1) {
+            const arg = ([]const u8)(paths[path_i]);
+            total_paths_len += arg.len;
+        }
     }
-}
 
-pub fn joinWindows(allocator: *Allocator, paths: ...) ![]u8 {
-    return mem.join(allocator, sep_windows, paths);
```

Comment:
why can't these functions use mem.join?

Reply:
Forgot to mention why this is important for paths.

If we are joining two paths (`/home/kristate/` and `zig`) with `/` then we have a problem that the path will become `/home/kristate/` `/` `zig` which is a valid result for the new `mem.join` implementation at PR #1808 but bad for `os.path.join` which should be `/home/kristate/zig`.

The test suite includes these tests, btw.